cmd/dogstatsd/subcommands/start/command.go

```diff
@@ -107,6 +111,22 @@ func RunDogstatsdFct(cliParams *CLIParams, defaultConfPath string, defaultLogFil
 		dogstatsd.Bundle,
 		forwarder.Bundle,
 		fx.Provide(defaultforwarder.NewParams),
+		// workloadmeta setup
+		collectors.GetCatalog(),
+		fx.Provide(func(config config.Component) workloadmeta.Params {
+			var catalog workloadmeta.AgentType
+			if flavor.GetFlavor() == flavor.ClusterAgent {
```

Comment:
I think we should only have the node agent catalog here, only if `dogstatsd_origin_detection` is provided

Reply:
Sorry, can you elaborate a little bit; I'm not sure I understand the condition you suggest?

Reply:
I think this code path is only used in the Node-Agent. Moreover it was protected by a config parameter `dogstatsd_origin_detection`